modules/connectors/hay_post/karrio/providers/hay_post/shipment/create.py

```diff
@@ -0,0 +1,119 @@
+import typing
+import karrio.lib as lib
+import karrio.core.units as units
+import karrio.core.models as models
+import karrio.providers.hay_post.error as error
+import karrio.providers.hay_post.utils as provider_utils
+import karrio.providers.hay_post.units as provider_units
+import karrio.schemas.hay_post.order_create_request as hay_post
+import karrio.schemas.hay_post.order_create_response as hay_post_response
+
+
+def parse_shipment_response(
+    _response: lib.Deserializable[dict],
+    settings: provider_utils.Settings,
+) -> typing.Tuple[models.ShipmentDetails, typing.List[models.Message]]:
+    response = _response.deserialize()
+
+    messages = []
+    shipment = []
+    if response.get('key') is None:
+        shipment = _extract_details(response, settings)
+    else:
+        messages = error.parse_error_response(response, settings)
+
+    return shipment, messages
+
+
+def _extract_details(
+    data: dict,
+    settings: provider_utils.Settings,
+) -> models.ShipmentDetails:
+    shipment = lib.to_object(hay_post_response.OrderCreateResponseType, data)
+    return models.ShipmentDetails(
+        carrier_id=settings.carrier_id,
+        carrier_name=settings.carrier_name,
+        tracking_number=shipment.barcode,
+        shipment_identifier=str(shipment.id),
+        docs=models.Documents(
+            label="No label...",
```

Comment:
No labels is returned?

Reply:
There is no label; the API returns a barcode and postal code.